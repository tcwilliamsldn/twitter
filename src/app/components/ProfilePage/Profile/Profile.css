.profile {
    background-color: #ffffff;
    margin: 12px;

    @media (width >= 768px) {
        width: 420px;
        flex-shrink: 0;
    }
}

.banner {
    height: 100px;
    background-color: #306aff;
}

.details {
    padding: 16px;
}

.personal {
    display: flex;
    margin-bottom: -28px;
}

.avatar {
    width: 96px;
    height: 96px;
    flex-shrink: 0;
    position: relative;
    border-radius: 50%;
    border: #ffffff solid 3px;
    top: -66px;
}

.names {
    margin-left: 16px;
    overflow: hidden;

    & > h1 {
        font-weight: bold;
        font-size: 24px;
        padding: 0;
        margin: 0;
        overflow: hidden;
        white-space: nowrap;
        text-overflow: ellipsis;
    }

    & > h2 {
        color: #737373;
        font-size: 18px;
        padding: 0;
        margin: 0;
        font-weight: normal;
    }
}

.stats {
    display: flex;
    position: relative;
    justify-content: space-between;
    width: 80%;
}
