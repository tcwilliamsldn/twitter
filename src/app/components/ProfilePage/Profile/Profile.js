import React from 'react';
import styles from './Profile.css';
import Statistic from './Statistic/Statistic';
import enlargeImage from '../../../utils/enlargeImage';

const Profile = props => {
    const {
        screen_name,
        name,
        profile_image_url,
        tweets_count,
        followers_count,
        following
    } = props;

    return (
        <div className={styles.profile}>
            <div className={styles.banner} />
            <section className={styles.details}>
                <div className={styles.personal}>
                    <img
                        alt={`Profile picture of ${name}`}
                        className={styles.avatar}
                        src={enlargeImage(profile_image_url)}
                    />
                    <div className={styles.names}>
                        <h1>{name}</h1>
                        <h2>@{screen_name}</h2>
                    </div>
                </div>
                <div className={styles.stats}>
                    <Statistic title="Tweets" value={tweets_count} />
                    <Statistic title="Followers" value={followers_count} />
                    <Statistic title="Following" value={following} />
                </div>
            </section>
        </div>
    );
};

export default Profile;
