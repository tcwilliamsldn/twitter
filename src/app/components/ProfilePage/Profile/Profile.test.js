import Profile from './Profile';
import React from 'react';
import { shallow } from 'enzyme';

describe('Profile', function() {
    const props = {
        screen_name: 'tom',
        name: 'Tom',
        profile_image_url: 'http://example.com/image.jpg',
        tweets_count: 123,
        followers_count: 123,
        following: 123
    };

    it('renders a users profile', function() {
        const component = shallow(<Profile {...props} />);
        expect(component).toMatchSnapshot();
    });

    it('resizes the profile image when the image URL contains "normal"', function() {
        const component = shallow(
            <Profile
                {...props}
                profile_image_url="http://example.com/imagenormal.jpg"
            />
        );
        expect(component).toMatchSnapshot();
    });
});
