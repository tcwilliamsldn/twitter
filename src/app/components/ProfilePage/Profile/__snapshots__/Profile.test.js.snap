// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile renders a users profile 1`] = `
<div
  className="profile"
>
  <div
    className="banner"
  />
  <section
    className="details"
  >
    <div
      className="personal"
    >
      <img
        alt="Profile picture of Tom"
        className="avatar"
        src="http://example.com/image.jpg"
      />
      <div
        className="names"
      >
        <h1>
          Tom
        </h1>
        <h2>
          @
          tom
        </h2>
      </div>
    </div>
    <div
      className="stats"
    >
      <Statistic
        title="Tweets"
        value={123}
      />
      <Statistic
        title="Followers"
        value={123}
      />
      <Statistic
        title="Following"
        value={123}
      />
    </div>
  </section>
</div>
`;

exports[`Profile resizes the profile image when the image URL contains "normal" 1`] = `
<div
  className="profile"
>
  <div
    className="banner"
  />
  <section
    className="details"
  >
    <div
      className="personal"
    >
      <img
        alt="Profile picture of Tom"
        className="avatar"
        src="http://example.com/image200x200.jpg"
      />
      <div
        className="names"
      >
        <h1>
          Tom
        </h1>
        <h2>
          @
          tom
        </h2>
      </div>
    </div>
    <div
      className="stats"
    >
      <Statistic
        title="Tweets"
        value={123}
      />
      <Statistic
        title="Followers"
        value={123}
      />
      <Statistic
        title="Following"
        value={123}
      />
    </div>
  </section>
</div>
`;
